### PR TITLE
[FLINK-6702] fix SIGABRT during GC in the CEP unit tests

### PR DESCRIPTION
**Pull request by NicoK**

### CODE REVIEW

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPOperatorTest.java [hunk 1]
```diff
@@ -654,6 +654,7 @@ public boolean filter(Event value) throws Exception {
 						Lists.newArrayList(startEvent2, endEvent, middleEvent3)
 				)
 		);
+		harness.close();
```
Comment (zentol) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPOperatorTest.java]: I think we should do this in a finally block, and modify other tests to do the same.

Comment (NicoK) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPOperatorTest.java]: ok, did that - would have been nice to use try-with-resources but I didn't want to change too much and an operator's `close()` method does allow throwing `Exception`s and so should `AbstractStreamOperatorTestHarness#close()` (probably)

**Comment by zentol** [comment]
Looks good. I will merge this after #3976 to avoid merge conflicts.

**Comment by zentol** [comment]
I'll merge this now.

**Pull request closed without merging**